src/wp-includes/script-loader.php

```diff
@@ -294,6 +294,9 @@ function wp_default_packages_scripts( $scripts ) {
 			case 'wp-edit-post':
 				array_push( $dependencies, 'media-models', 'media-views', 'postbox', 'wp-dom-ready' );
 				break;
+			case 'wp-preferences':
+				array_push( $dependencies, 'wp-preferences-persistence' );
+				break;
```

Comment:
Don’t think this `case` is necessary as `script-loader-packages.php` contains  everything needed for WordPress to know that `wp-preferences` depends on `wp-preferences-persistence`. This `switch`  is for dependencies that can’t be derived from the `package.json`.

Reply:
I don't think it can be derived from the package.json.

The two packages intentionally have no coupling, as one has WordPress specific code and the other doesn't.

Reply:
Oh! I see. And we want it so that when `wp-preferences` is enqueued it automatically has a persistence layer set up. OK then 👍